File: packages/core/src/test/user/short-user.spec.ts
import { ObjectType } from "../../lib/core/object-type.js";
import { $ShortUser } from "../../lib/user/short-user.js";
import { registerJsonIoTests } from "../helpers.js";

describe("ShortUser", function () {
  registerJsonIoTests(
    $ShortUser,
    "core/user/short-user",
    new Map([
      [
        "demurgos",
        {
          type: ObjectType.User,
          id: "9f310484-963b-446b-af69-797feec6813f",
          displayName: {
            current: {
              value: "Demurgos",
            },
          },
        },
      ],
    ])
  );
});

File: clients/ruby/spec/user/short_user_spec.rb
# typed: false
# frozen_string_literal: true

RSpec.describe Etwin::User::ShortUser do
  items = SerializationTestItem.from_test_dir(
    'core/user/short-user',
    {
      'demurgos' => Etwin::User::ShortUser.new(
        Etwin::User::UserId.new('9f310484-963b-446b-af69-797feec6813f'),
        Etwin::User::UserDisplayNameVersions.new(
          Etwin::User::UserDisplayNameVersion.new(
            Etwin::User::UserDisplayName.new('Demurgos')
          )
        )
      )
    }
  )

  items.each do |item|
    it "reads #{item.name}" do
      actual = Etwin::User::ShortUser.from_json(item.json)
      expect(actual).to eq(item.value)
    end
  end
end
